core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -605,6 +645,77 @@ case class InputPSBTMap(elements: Vector[InputPSBTRecord]) extends PSBTMap {
       .filter(element => PSBTInputKeyId.fromByte(element.key.head) == key)
       .asInstanceOf[Vector[T]]
   }
+
+  def isFinalized: Boolean =
+    getRecords(FinalizedScriptSigKeyId).nonEmpty || getRecords(
+      FinalizedScriptWitnessKeyId).nonEmpty
+
+  /**
+    * Takes the InputPSBTMap returns a UTXOSpendingInfo
+    * that can be used to sign the input
+    * @param txIn The transaction input that this InputPSBTMap represents
+    * @param signers Signers that will be used to sign the input
+    * @return A corresponding UTXOSpendingInfo
+    */
+  def toUTXOSpendingInfo(
+      txIn: TransactionInput,
+      signers: Seq[Sign]): UTXOSpendingInfo = {
+    toUTXOSpendingInfo(txIn, signers, ConditionalPath.NoConditionsLeft)
+  }
+
+  /**
+    * Takes the InputPSBTMap returns a UTXOSpendingInfo
+    * that can be used to sign the input
+    * @param txIn The transaction input that this InputPSBTMap represents
+    * @param signers Signers that will be used to sign the input
+    * @param conditionalPath Path that should be used for the script
+    * @return A corresponding UTXOSpendingInfo
+    */
+  def toUTXOSpendingInfo(
+      txIn: TransactionInput,
+      signers: Seq[Sign],
+      conditionalPath: ConditionalPath): UTXOSpendingInfo = {
+    require(!isFinalized, s"Cannot update an InputPSBTMap that is finalized")
+
+    val outPoint = txIn.previousOutput
+
+    val witVec = getRecords[WitnessUTXO](WitnessUTXOKeyId)
+    val txVec = getRecords[NonWitnessOrUnknownUTXO](NonWitnessUTXOKeyId)
+
+    val output = if (witVec.size == 1) {
+      witVec.head.spentWitnessTransaction
+    } else if (txVec.size == 1) {
+      val tx = txVec.head.transactionSpent
+      tx.outputs(txIn.previousOutput.vout.toInt)
+    } else {
+      throw new UnsupportedOperationException(
+        "Not enough information in the InputPSBTMap to get a valid UTXOSpendingInfo")
+    }
+
+    val redeemScriptVec = getRecords[RedeemScript](RedeemScriptKeyId)
+    val redeemScriptOpt =
+      if (redeemScriptVec.size == 1) Some(redeemScriptVec.head.redeemScript)
```

Comment:
nit: I think you could use `redeemScriptVec.headOption.map(_.redeemScript)` instead

Reply:
Doesn't work :( the return type is `Any`